aspnetcore/grpc/loadbalancing.md

```diff
@@ -197,30 +197,27 @@ A resolver:
 * Can optionally provide a service configuration.
 
 ```csharp
-public class FileResolver : Resolver
+public class FileResolver : AsyncResolver
 {
     private readonly Uri _address;
-    private Action<ResolverResult> _listener;
+    private readonly int _port;
 
-    public ExampleResolver(Uri address)
+    public ExampleResolver(Uri address, int defaultPort, ILoggerFactory loggerFactory)
+        : base(loggerFactory)
     {
         _address = address;
+        _port = defaultPort;
     }
 
-    public override async Task RefreshAsync(CancellationToken cancellationToken)
+    public override async Task ResolveAsync(CancellationToken cancellationToken)
     {
         // Load JSON from a file on disk and deserialize into endpoints.
         var jsonString = await File.ReadAllTextAsync(_address.LocalPath);
         var results = JsonSerializer.Deserialize<string[]>(jsonString);
-        var addresses = results.Select(r => new DnsEndPoint(r, 80));
+        var addresses = results.Select(r => new BalancerAddress(r, _port)).ToArray();
```

Comment:
Why do we need `ToArray` here now?

Reply:
ForResult takes an `IReadOnlyList<BalancerAddress>`.